File: Haskell/Programming/Haskell/weeklies/week1.py
def suma(a, b, c):
    possible = False
    if a + b == c:
        possible == True
    elif a + c == b:
        possible == True
    elif b + c == a:
        possible == True
    return possible

def popCount(a) -> int:
    accumulator = 0
    while a > 0:
        accumulator = a % 2
        a = a / 2
    return accumulator


from collections import Counter 
def checksum(strings) -> int:
    twos = 0
    threes = 0
    for string in strings:
        freqs = Counter(string)
        for freq in freqs:
            if freqs[freq] == 2:
                twos += 1
            elif freqs[freq] == 3:
                threes += 1

    print(twos * threes)


test = ["abcdef", "bababc", "abcdee", "aaa", "bb"]

checksum(test)

File: Haskell/Programming/Haskell/weeklies/week1.hs
import Data.Bits

true :: a -> b -> a
true a b = a

false :: a -> b -> b
false a b = b

if' :: Bool -> a -> a -> a
if' a
    | a = true
    | otherwise = false

suma a b c = if' (a + b == c || a + c == b || b + c == a) "YES" "NO"

populationCount :: (Bits a) => a -> Int
populationCount = popCount

checksum = undefined
